### ISSUE TITLE
When the code will be released

### ISSUE THREAD

**Issue by liberty-hit** [CVI-SZU/CLIMS#1]
Great jobs, btw, I wonder when will you release the full training code

**Comment by liberty523422842** [comment]
By the way, there is a typo in your table, the supervision for works including yours should be I (Image) not S (Saliency) I think

**Comment by Sierkinhane** [comment]
There are a lot of things to prepare recently and I hope to release the code in the coming months.

**Comment by Sierkinhane** [comment]
> By the way, there is a typo in your table, the supervision for works including yours should be I (Image) not S (Saliency) I think

Table 2?

**Comment by liberty-hit** [comment]
> > By the way, there is a typo in your table, the supervision for works including yours should be I (Image) not S (Saliency) I think
> 
> Table 2?

yes

**Comment by liberty-hit** [comment]
I'm trying to reproduce your result, but the training loss doesn't decrease at all...
In the Sec 3.3, you said the loss is log(s), in which s is a cosine similarity, so $s \in [-1, 1]$, and log(s) could be nan, did you really mean this, or you use other format losses

**Comment by liberty-hit** [comment]
> 

When I trying to reproduce, I use logit_scale to scale  cosine similarity to logits \in [-100/7, 100/7] and then use logsigmoid to  compute the loss, I wonder whether I use the correct format.

**Comment by Sierkinhane** [comment]
> > 
> 
> When I trying to reproduce, I use logit_scale to scale cosine similarity to logits \in [-100/7, 100/7] and then use logsigmoid to compute the loss, I wonder whether I use the correct format.

I just use the cosine similarity between two vectors, norm(v1) * norm(v2).T

**Comment by Sierkinhane** [comment]
> > 
> 
> When I trying to reproduce, I use logit_scale to scale cosine similarity to logits \in [-100/7, 100/7] and then use logsigmoid to compute the loss, I wonder whether I use the correct format.

In my experiments, the loss slightly decreased but the activation map gradually approached the target category.

**Comment by liberty-hit** [comment]
> > > 
> > 
> > 
> > When I trying to reproduce, I use logit_scale to scale cosine similarity to logits \in [-100/7, 100/7] and then use logsigmoid to compute the loss, I wonder whether I use the correct format.
> 
> I just use the cosine similarity between two vectors, norm(v1) * norm(v2).T

but how to compute log on the cosine similarity less than 0

**Comment by Sierkinhane** [comment]
clamp (0, 1)

**Comment by liberty-hit** [comment]
> clamp (0, 1)

ok, I'll try that formation, thanks for your reply

**Comment by Sierkinhane** [comment]
I hope to release the code in the coming months :-)

**Comment by liberty-hit** [comment]
> I hope to release the code in the coming months :-)

Great!, by the way, could you please provide the pre-defined cbs set

**Comment by Sierkinhane** [comment]
{'boat': ['a photo of river', 'a photo of water', 'a photo of lake', 'a photo of sea', 'a photo of building'],
            'train': ['a photo of railroad', 'a photo of railway', 'a photo of branches', 'a photo of tree'], You can try this one.

**Comment by liberty-hit** [comment]
> {'boat': ['a photo of river', 'a photo of water', 'a photo of lake', 'a photo of sea', 'a photo of building'], 'train': ['a photo of railroad', 'a photo of railway', 'a photo of branches', 'a photo of tree'], You can try this one.

Nice

**Comment by liberty-hit** [comment]
@Sierkinhane Sry to bother you again, could you please provide the finetuned CLIP model, I've finetuned the CLIP model using RN50 as backbone and got ~96% accuracy on VOC12 train_aug, but still failed to train CLIMS even only using loss_otm, the miou of CAMs on train set is only 19%

**Comment by liberty-hit** [comment]
I've found the code to finetune CLIP (https://github.com/openai/CLIP/issues/83), I'll try it

**Comment by liberty-hit** [comment]
But I wonder whether I use F.multilabel_soft_margin_loss and treat finetuning CLIP as a multi-label classification task is correct

**Comment by Sierkinhane** [comment]
Details were provided in email. : )

**Comment by Sierkinhane** [comment]
@liberty-hit hello, the source code is released now.
